src/hotspot/share/prims/jvmtiAgentList.cpp

```diff
@@ -220,6 +220,22 @@ void JvmtiAgentList::unload_agents() {
   }
 }
 
+// Return true if a library is a loaded agent library
+bool JvmtiAgentList::is_loaded(void* os_lib) {
+  JvmtiAgentList::Iterator it = JvmtiAgentList::agents();
+  while (it.has_next()) {
+    JvmtiAgent* const agent = it.next();
+    if (os_lib == nullptr) {
+       if (agent->is_static_lib() && agent->is_loaded()) {
+         return true;
+       }
```

Comment:
load_agent_from_executable has a comment to explain how statically linked agents are started, that's why it needs to use agent->is_static_lib().&& agent->is_loaded() here. There isn't currently a way to test this but there is other work going to support static builds so it might be possible to write some automated tests at that point.

Reply:
I've updated this so that is_loaded is split into one function to test if a statically linked agent is already loaded and another for agent libraries. That might be clearer to understand.